### PR TITLE
#740 Add HTTP integration tests

### LINKED ISSUE #740: Http Integration Tests: Create tests to support HTTP(7)
This issue involves creating the below test cases for HTTP as well:

**Source**: integration_tests/commands/async/

- hsetnx_test.go
- hstrlen_test.go
- hyperloglog_test.go
- incr_by_float_test.go
- incr_test.go

Steps:
1) Create the corresponding tests for these commands in integration_tests/commands/http.
2) Use set_test.go as a reference, and how to alter the input and output to match JSON structures from RESP.
3) Run the tests and fix issues with command parsing in HTTP.

If you need to do anything **custom** to support HTTP, let @pratikpandey21 @lucifercr07 know.

### PR DESCRIPTION
**Pull request by hgupta12**
Resolves #740 and #929

### CODE REVIEW

**Comment by lucifercr07** [comment]
@hgupta12 please fix lint errors.

**Comment by JyotinderSingh** [comment]
> @hgupta12 please fix lint errors.

Looks like the lint errors are unrelated and can be ignored, they are fixed on main.

File: integration_tests/commands/async/hyperloglog_test.go [hunk 1]
```diff
@@ -51,12 +51,6 @@ func TestHyperLogLogCommands(t *testing.T) {
 				"PFMERGE NON_EXISTING_SRC_KEY", "PFCOUNT NON_EXISTING_SRC_KEY"},
 			expected: []interface{}{"OK", int64(0)},
 		},
-		{
-			name: "PFMERGE with srcKey non-existing",
-			commands: []string{
-				"PFMERGE NON_EXISTING_SRC_KEY", "PFCOUNT NON_EXISTING_SRC_KEY"},
-			expected: []interface{}{"OK", int64(0)},
-		},
```
Comment (JyotinderSingh) [inline integration_tests/commands/async/hyperloglog_test.go:59]: Why is this removed?

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for helping improve the test coverage of the HTTP test suite @hgupta12! The changes look good for the most part. Have left a small comment.

File: integration_tests/commands/async/hyperloglog_test.go (hunk 1 above)
Comment (hgupta12) [reply integration_tests/commands/async/hyperloglog_test.go:59]: This test case is exactly same as the one above it. I guess it was copied by mistake.
